dawn/src/dawn/Optimizer/Driver.cpp

```diff
@@ -24,4 +66,255 @@ std::list<PassGroup> defaultPassGroups() {
+      passManager.pushBackPass<PassTemporaryType>();
+      // passManager.pushBackPass<PassFixVersionedInputFields>();
+      // validation check
+      passManager.pushBackPass<PassValidation>();
+      break;
+    case PassGroup::TmpToStencilFunction:
+      passManager.pushBackPass<PassTemporaryToStencilFunction>();
+      // validation check
+      passManager.pushBackPass<PassValidation>();
+      break;
+    case PassGroup::SetNonTempCaches:
+      passManager.pushBackPass<PassSetNonTempCaches>();
+      // this should not affect the temporaries but since we're touching them it would probably be
+      // a safe idea
+      passManager.pushBackPass<PassTemporaryType>();
+      passManager.pushBackPass<PassLocalVarType>();
+      // validation check
+      passManager.pushBackPass<PassValidation>();
+      break;
+    case PassGroup::SetCaches:
+      passManager.pushBackPass<PassSetCaches>();
+      // validation check
+      passManager.pushBackPass<PassValidation>();
+      break;
+    case PassGroup::SetBlockSize:
+      if(stencilInstantiationMap.begin()->second->getIIR()->getGridType() !=
+         ast::GridType::Unstructured) {
+        passManager.pushBackPass<PassSetBlockSize>();
+        // validation check
+        passManager.pushBackPass<PassValidation>();
+      } else {
+        DAWN_LOG(WARNING) << "PassSetBlockSize currently disabled for unstructured meshes!";
+      }
+      break;
+    case PassGroup::DataLocalityMetric:
+      // Plain diagnostics, should not even be a pass but is independent
+      passManager.pushBackPass<PassDataLocalityMetric>();
+      // validation check
+      passManager.pushBackPass<PassValidation>();
+      break;
+    case PassGroup::Parallel:
+      DAWN_ASSERT_MSG(false, "The parallel group is only valid for lowering to IIR.");
+    }
+  }
+  // Note that we need to run PassInlining here if serializing or using the Cuda codegen backend.
+  if(options.SerializeIIR) {
+    passManager.pushBackPass<PassInlining>(PassInlining::InlineStrategy::ComputationsOnTheFly);
+  }
+
+  //===-----------------------------------------------------------------------------------------
+
+  dawn::log::error.clear();
+  for(auto& stencil : stencilInstantiationMap) {
+    // Run optimization passes
+    auto& instantiation = stencil.second;
+
+    DAWN_LOG(INFO) << "Starting optimization and analysis passes for `" << instantiation->getName()
+                   << "` ...";
+    if(!passManager.runAllPassesOnStencilInstantiation(instantiation, options))
+      throw std::runtime_error("An error occurred.");
+
+    DAWN_LOG(INFO) << "Done with optimization and analysis passes for `" << instantiation->getName()
+                   << "`";
+
+    if(options.SerializeIIR) {
+      const IIRSerializer::Format serializationKind =
+          options.SerializeIIR ? IIRSerializer::parseFormatString(options.IIRFormat)
+                               : IIRSerializer::Format::Json;
```

Comment:
Have we ever used a serialization format different from json?

Reply:
I believe this is just defaulting to Json unless we specify the output format.

Reply:
I believe @twicki thinks we should keep the byte IO option. Thoughts @twicki?

Reply:
I would agree, no harm in keeping that option.